libs/node/utils/src/lib/setupShutdownHooks.ts

```diff
@@ -0,0 +1,40 @@
+import { logger } from '@island.is/logging'
+import { Server } from 'http'
+
+export const setupShutdownHooks = (
+  server: Server,
+  onShutdown?: () => Promise<void>,
+) => {
+  const terminationEvents = ['SIGHUP', 'SIGINT', 'SIGTERM']
+
+  // Make sure the server doesn't hang after parent process disconnects, eg when
+  // e2e tests are finished.
+  if (process.env.NX_INVOKED_BY_RUNNER === 'true') {
+    terminationEvents.push('disconnect')
+  }
+
+  let isShuttingDown = false
```

Comment:
~personal nit: I love to separate methods, classes, etc. from other code with a new line :)